src/oauth-provider.ts

```diff
@@ -1621,6 +1661,126 @@ class OAuthProviderImpl {
+   * @returns Response confirming revocation or error
+   */
+  private async revokeToken(body: any, env: any): Promise<Response> {
+    const token = body.token;
+    const tokenTypeHint = body.token_type_hint;
+
+    if (!token) {
+      return this.createErrorResponse('invalid_request', 'Token parameter is required');
+    }
+
+    // Validate token_type_hint before processing (RFC 7009)
+    if (tokenTypeHint && tokenTypeHint !== 'access_token' && tokenTypeHint !== 'refresh_token') {
+      return this.createErrorResponse('unsupported_token_type', 'Unsupported token type hint');
+    }
+
+    try {
+      // Parse the token to extract user ID and grant ID
+      const tokenParts = token.split(':');
+      if (tokenParts.length !== 3) {
+        // Invalid token format, but RFC 7009 requires 200 response even for invalid tokens
+        return new Response('', { status: 200 });
+      }
+
+      const [userId, grantId, _] = tokenParts;
+
+      // Determine what type of token this is by checking if it's an access token or refresh token
+      let isValidToken = false;
+
+      // If hint is provided, try that type first for efficiency
+      if (tokenTypeHint === 'access_token') {
+        isValidToken = await this.validateAccessToken(token, userId, grantId, env);
+      } else if (tokenTypeHint === 'refresh_token') {
+        isValidToken = await this.validateRefreshToken(token, userId, grantId, env);
+      }
+
+      // If hint didn't work or no hint provided, try both types
+      if (!isValidToken) {
+        isValidToken =
+          (await this.validateAccessToken(token, userId, grantId, env)) ||
+          (await this.validateRefreshToken(token, userId, grantId, env));
+      }
+
+      // If we found a valid token, revoke the entire grant
+      if (isValidToken) {
+        await this.createOAuthHelpers(env).revokeGrant(grantId, userId);
+      }
+
+      return new Response('', { status: 200 });
+    } catch (error) {
+      console.error('Token revocation error:', error);
+      return new Response('', { status: 200 });
+    }
+  }
+
+  /**
+   * Validates if a token is a valid access token
+   * @param token - The token to validate
+   * @param userId - The user ID extracted from the token
+   * @param grantId - The grant ID extracted from the token
+   * @param env - Cloudflare Worker environment variables
+   * @returns Promise<boolean> indicating if the token is valid
+   */
+  private async validateAccessToken(token: string, userId: string, grantId: string, env: any): Promise<boolean> {
+    try {
+      const accessTokenId = await generateTokenId(token);
+      const tokenKey = `token:${userId}:${grantId}:${accessTokenId}`;
+      const tokenData = await env.OAUTH_KV.get(tokenKey, { type: 'json' });
+
+      if (!tokenData) {
+        return false;
+      }
+
+      // Check if token is expired
+      const now = Math.floor(Date.now() / 1000);
+      return tokenData.expiresAt >= now;
+    } catch {
```

Comment:
I don't think we should catch exceptions and assume it means the token is invalid. An exception caught here almost certainly is due to some sort of bug in the code. Silently ignoring it and returning false means we won't ever find out about the bug -- we'll just have tokens that aren't correctly revoked.